docs/external-database.asciidoc

```diff
@@ -0,0 +1,50 @@
+= External database support
+
+Keycloak Operator allows you to use an external Postgresql database by modifying `keycloak-postgresql` Endpoints object.
+The Operator also reuses the same `keycloak-db-secret` Secret but will require two additional properties:
+
+* POSTGRES_EXTERNAL_ADDRESS - an IP address (or a Hostname) or the external database. This address needs
+  to be resolvable from Kubernetes cluster perspective.
+* POSTGRES_EXTERNAL_PORT - (Optional) A database port.
+
+The other properties remain the same for both operating modes (hosted Postgresql and an external one).
+
+Here's an example of such secret (note that values are Base64 encoded, see https://kubernetes.io/docs/concepts/configuration/secret/#using-secrets[Kubernetes Secrets manual]:
+
+.`keycloak-db-secret` Secret
+```yaml
+apiVersion: v1
+data:
+  POSTGRES_DATABASE: cm9vdA==
+  POSTGRES_EXTERNAL_ADDRESS: MTcyLjE3LjAuMw==
+  POSTGRES_EXTERNAL_PORT: NTQzMg==
+  POSTGRES_HOST: a2V5Y2xvYWstcG9zdGdyZXNxbA==
```

Comment:
It is unfortunate the end user needs to supply this as it is the service name that is still created by our operator.

Reply:
Actually `POSTGRES_HOST` is not needed by our Operator. This is needed by Integreately Backups :disappointed: See https://github.com/integr8ly/backup-container-image/blob/master/image/tools/lib/component/postgres.sh#L16

Personally, I would love to remove it but then, backups wouldn't work :disappointed:

Reply:
Thanks @slaskawi . What I was talking about here was that the HOST from what I can see in the code is the name of the database service we create - https://github.com/keycloak/keycloak-operator/blob/master/pkg/model/database_secret.go#L53

This means the end user will need to know this secret name. I think for now it is fine to just explicitly list that in the docs.

Reply:
ooh I get what you mean. Let me update the docs.

Reply:
Thanks @slaskawi

Reply:
looks great